lib/chromic_pdf/api.ex

```diff
@@ -60,6 +61,32 @@ defmodule ChromicPDF.API do
     |> Keyword.put_new(source_type, source)
   end
 
+  defp rewrite_wait_for(opts) do
+    opts
+    |> Keyword.pop(:wait_for)
+    |> do_rewrite_wait_for()
+  end
+
+  defp do_rewrite_wait_for({nil, opts}), do: opts
+
+  defp do_rewrite_wait_for({%{selector: selector, attribute: attribute}, opts}) do
+    if Keyword.has_key?(opts, :evaluate) do
+      raise("wait_for option cannot be combined with evaluate option")
+    end
+
+    Keyword.put(opts, :evaluate, %{
+      expression: """
+      const waitForAttribute = async (selector, attribute) => {
+        while (!document.querySelector(selector).hasAttribute(attribute)) {
+          await new Promise(resolve => requestAnimationFrame(resolve));
+        }
+      };
+
+      waitForAttribute('#{selector}', '#{attribute}');
```

Comment:
Would it make sense to put such half js things into a `js.eex` template, so editors can syntax highlight them better?

The module is already quite big, so maybe putting some of it into a separate file would be nice :)

Reply:
i totally agree on the "too big" part, only having troubles identifying a suitable name / concept for a different module. Something along the lines of `BrowserOptions` or so (to keep everything in `chrome_export`), but it whenever I move these functions somewhere else they feel really out of place by themselves. Weird functionality on some obscure `opts` dict...

Reply:
I moved it into an eex template but left it in the same file 😄  Probably not what you wanted, but closer to it. If you can figure out a good name / concept to extract from the `API` module, pls go ahead